src/main/java/edu/ksu/canvas/interfaces/UserWriter.java

```diff
@@ -25,5 +25,15 @@
      * @throws InvalidOauthTokenException When the supplied OAuth token is not valid
      * @throws IOException When there is an error communicating with Canvas
      */
-     Optional<User> updateUser(User user) throws InvalidOauthTokenException, IOException;
+    Optional<User> updateUser(User user) throws InvalidOauthTokenException, IOException;
+
+    /**
+     * Update user information using the supplied URL parameter options.
+     * @param options the options to send across
+     * @return an optional user, if any records exist or were updated.
+     * @throws InvalidOauthTokenException When the supplied OAuth token is not valid
+     * @throws IOException When there is an error communicating with Canvas
+     * @see UpdateUserOptions
+     */
+    Optional<User> updateUser(UpdateUserOptions options) throws InvalidOauthTokenException, IOException;
```

Comment:
Why make this updateUser method when you can set all of these properties on a `User` object and then user the other updateUser method?

Reply:
Oh, I think I answered my own question. It looks like the existing `updateUser` method doens't actually work correctly. It is only setting two properties instead of serializing the User object to JSON like other update methods do. I should probably look at fixing that...

Reply:
Right - I believe at the time I didn't want to cause any potential breakage with existing behavior.  We can definitely unify it into one object once the work to fix up the `updateUser` method is done.